src/openvic-simulation/economy/Producer.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include "openvic-simulation/economy/Good.hpp"
+#include "openvic-simulation/economy/ProductionType.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/utility/Getters.hpp"
+
+namespace OpenVic {
+	class Producer {
+	public:
+		ProductionType const& PROPERTY(production_type);
```

Comment:
Tbh I struggle with understand and picking the PROPERTY stuff. It's all very custom.
I agree they should be protected and will change it. As for explicit default, I like it that way. You instantly see the accessibility instead of having to look up/remember the default setting of the container.